hathor/simulator/simulator.py

```diff
@@ -127,52 +126,45 @@ def stop(self) -> None:
-                    enable_sync_v1: bool = True, enable_sync_v2: bool = True,
-                    soft_voided_tx_ids: Optional[Set[bytes]] = None) -> HathorManager:
-        # TODO Refactor to use Builder.
-        assert self._started
-        if network is None:
-            network = self._network
+    def create_peer(
+        self,
+        network: Optional[str] = None,
+        peer_id: Optional[PeerId] = None,
+        enable_sync_v1: bool = True,
+        enable_sync_v2: bool = True,
+        soft_voided_tx_ids: Optional[Set[bytes]] = None,
+        full_verification: bool = True
+    ) -> HathorManager:
+        assert self._started, 'Simulator is not started.'
+        assert peer_id is not None  # XXX: temporary, for checking that tests are using the peer_id
 
         wallet = HDWallet(gap_limit=2)
         wallet._manually_initialize()
 
-        pubsub = PubSubManager(self._clock)
-
-        if soft_voided_tx_ids is None:
-            soft_voided_tx_ids = set()
-        consensus_algorithm = ConsensusAlgorithm(soft_voided_tx_ids, pubsub=pubsub)
-
-        assert peer_id is not None  # XXX: temporary, for checking that tests are using the peer_id
-        if peer_id is None:
-            peer_id = PeerId()
-        tx_storage = TransactionMemoryStorage()
-        manager = HathorManager(
-            self._clock,
-            pubsub=pubsub,
-            consensus_algorithm=consensus_algorithm,
-            peer_id=peer_id,
-            network=network,
-            wallet=wallet,
-            enable_sync_v1=enable_sync_v1,
-            enable_sync_v2=enable_sync_v2,
-            tx_storage=tx_storage,
-            rng=Random(self.rng.getrandbits(64)),
-        )
-
-        manager.reactor = self._clock
-        manager._full_verification = True
-        manager.start()
+        artifacts = Builder() \
+            .set_reactor(self._clock) \
+            .set_peer_id(peer_id or PeerId()) \
+            .set_network(network or self._network) \
+            .set_wallet(wallet) \
+            .set_rng(Random(self.rng.getrandbits(64))) \
+            .set_tx_storage(TransactionMemoryStorage()) \
+            .set_enable_sync_v1(enable_sync_v1) \
+            .set_enable_sync_v2(enable_sync_v2) \
+            .set_full_verification(full_verification) \
+            .set_soft_voided_tx_ids(soft_voided_tx_ids or set()) \
+            .build()
+
+        artifacts.manager.start()
         self.run_to_completion()
 
         # Don't use it anywhere else. It is unsafe to generate mnemonic words like this.
         # It should be used only for testing purposes.
         m = Mnemonic('english')
         words = m.to_mnemonic(self.rng.randbytes(32))
         self.log.debug('randomized step: generate wallet', words=words)
-        wallet.unlock(words=words, tx_storage=manager.tx_storage)
-        return manager
+        wallet.unlock(words=words, tx_storage=artifacts.manager.tx_storage)
```

Comment:
You can use `artifacts.tx_storage` instead of `artifacts.manager.tx_storage`.

Reply:
Thanks! Done in https://github.com/HathorNetwork/hathor-core/pull/561/commits/bcd0b3e99b73fa96f64ca92141d54d63732b6795

Reply:
Thanks!